odl/diagnostics/operator.py

```diff
@@ -75,29 +86,27 @@ def norm(self):
         Wikipedia article on `Operator norm
         <https://en.wikipedia.org/wiki/Operator_norm>`_.
         """
-        print('\n== Calculating operator norm ==\n')
-
-        operator_norm = 0.0
-        for _, x in samples(self.operator.domain):
-            result = self.operator(x)
-            x_norm = x.norm()
-            estimate = 0 if x_norm == 0 else result.norm() / x_norm
+        self.log('\n== Calculating operator norm ==\n')
 
-            operator_norm = max(operator_norm, estimate)
+        operator_norm = max(power_method_opnorm(self.operator, niter=2,
+                                                xstart=x)
+                            for name, x in samples(self.operator.domain)
+                            if name != 'Zero')
 
-        print('Norm is at least: {}'.format(operator_norm))
+        self.log('Norm is at least: {}'.format(operator_norm))
         self.operator_norm = operator_norm
         return operator_norm
 
     def self_adjoint(self):
-        """Verify (Ax, y) = (x, Ay)."""
-        name = 'Verifying the identity (Ax, y) = (x, Ay)'
-
+        """Verify ``<Ax, y> == <x, Ay>``."""
         left_inner_vals = []
         right_inner_vals = []
 
-        with FailCounter(name, 'error = ||(Ax, y) - (x, Ay)|| / '
-                         '||A|| ||x|| ||y||') as counter:
+        with FailCounter(
+                test_name='Verifying the identity <Ax, y> = <x, Ay>',
+                err_msg='error = |<Ax, y> - <x, Ay>| / ||A|| ||x|| ||y||',
+                verbose=self.verbose) as counter:
```

Comment:
Now, I'm going to ask you to rewrite this again. Instead of passing verbose here, why not create an argument "out" or something and then pass `self.log`. That way the failcounter can write to the actual log?